### PR TITLE
Feat/am UI/1091 modal husker siste søk når man går tilbake fra resu

### PR DESCRIPTION
**Pull request by sonwit**
## Description

Husker søktekst og filter i modal mellom søk og detaljsiden  
Legger på snack-bar vet fullført delegering

## Related Issue(s)
- #1091

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx [hunk 1]
```diff
@@ -140,9 +144,21 @@ export const ResourceInfo = ({ resource, toParty, onDelegate }: ResourceInfoProp
...
       };
 
-      delegateRights(delegationInput).then(() => {
-        onDelegate();
-      });
+      delegateRights(delegationInput)
+        .then(() => {
+          openSnackbar({
+            message: t('delegation_modal.success_message', { name: toParty.name }),
+            variant: SnackbarMessageVariant.Success,
```
Comment (allinox) [inline src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx:151]:
```suggestion
            variant: SnackbarMessageVariant.Success,
            duration: SnackbarDuration.long,
```
I personally think the normal duration is a bit too short. Consider changing it to long.

File: src/localizations/no_nn.json [hunk 2]
```diff
@@ -208,12 +208,14 @@
     "resource_type_text": "Enkelttjeneste",
     "delete_ingleRight_confirm_message": "Er du sikker på at du vil slette denne?",
     "delete_singleRight_success_message": "Tilgang til tjenesten: {{servicename}} er slettet",
-    "delete_singleRight_error_message": "Kunne ikke slette tilgang til tjenesten {{servicename}}"
+    "delete_singleRight_error_message": "Kunne ikkje slette tilgang til tjenesten {{servicename}}"
```
Comment (allinox) [inline src/localizations/no_nn.json:211]:
```suggestion
    "delete_singleRight_error_message": "Kunne ikkje slette tilgang til tenesta {{servicename}}"
```

**Review by allinox — COMMENTED** [review]
Looks great! 😄
The modal context seems to work well and it's great to see snackbars popping up when delegating.

I just have a couple of comments that I'd like to hear your opinion on

File: src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx (hunk 1 above)
Comment (sonwit) [reply src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx:151]: Agreed 👍

File: src/features/amUI/userRightsPage/DelegationModal/ResourceSearch.tsx [hunk 3]
```diff
@@ -193,6 +202,7 @@ export const ResourceSearch = ({ onSelection }: ResourceSearchProps) => {
           <Search
```
[the remark this answers is not in the document]
Comment (allinox) [reply src/features/amUI/userRightsPage/DelegationModal/ResourceSearch.tsx:193]: Great! It works now! 😄

**Review by allinox — APPROVED** [review]
🙌✨🚀

**Pull request merged**